liualgotrader/trading/base.py

```diff
@@ -91,6 +93,43 @@ async def run(self) -> Optional[asyncio.Task]:
 
     async def close(self):
         ...
+      
+    def get_broker(self):
+        return config.broker
+        
+    def broker_working_limits(self, start_date, end_date):
+        market_opens, market_closes = get_market_schedule()
+        hours = market_closes - market_opens     
+        
+        days = len(get_trading_days(start_date, end_date).index)
+        return hours, days
+        
+    def max_data_points_per_load(self):
```

Comment:
@ksilo this should not be here! the whole purpose of object-oriented inheritance is that you keep the base generic, and each implementation (e.g. Alpaca, Gemini etc) return their own values. You need to add the specific functions to the respective trading implementations (this is missing in your repo).